specifyweb/frontend/js_src/lib/components/Attachments/__tests__/utils.test.ts

```diff
@@ -0,0 +1,32 @@
+import { requireContext } from '../../../tests/helpers';
+import type { RA } from '../../../utils/types';
+import { schema } from '../../DataModel/schema';
+import type { SpecifyModel } from '../../DataModel/specifyModel';
+import {
+  allTablesWithAttachments,
+  attachmentRelatedTables,
+  getAttachmentRelationship,
+} from '../utils';
+
+requireContext();
+
+test('attachmentRelatedTables', () =>
+  expect(attachmentRelatedTables()).toMatchSnapshot());
+
+test('allTablesWithAttachments', () =>
+  expect(allTablesWithAttachments()).toMatchSnapshot());
+
+test('getAttachmentRelationship', () => {
```

Comment:
(optional) could use `theories` util or https://jestjs.io/docs/api#testeachtablename-fn-timeout here

Reply:
I had initially used `theories` and `test.each` here, but in both cases it seems the tests were being run in an environment without access to the schema. 
In other words, they were missing the context from the `requireContext()` call.

Reply:
ah I see
that is because you were using `schema.models.DNASequence` in the arguments array
the arguments array are created as soon as the file is opened - thus, before schema is loaded
where as the `test()` is only called after schema is loaded, thus accessing `schema.models.DNASequence` inside of it is not a problem

two solutions to that (the way you have it now is fine, just mentioning for your awareness):
- in the test file, create a small wrapper for getAttachmentRelationship - the wrapper would accept table name as a string, and resolve that to the table (since this resolving would only happen when the test is being run, the schema is loaded by then)
- modify `theories` to support passing a function that would be called by `theories` to get the arguments, rather than providing arguments right away. similar to have `overrideAjax` supports that - see this usage for example https://github.com/specify/specify7/blob/3402921830119d428802b74b0d8d9a38305efa53/specifyweb/frontend/js_src/lib/components/QueryBuilder/__tests__/fromTree.test.ts#L35

Reply:
Fixed in https://github.com/specify/specify7/pull/4414/commits/693113be83c3fe55ab27bc31c56394af42215215